### ISSUE TITLE
dollarmath matches aggressively

### ISSUE THREAD

**Issue by jpmckinney** [executablebooks/MyST-Parser#305]
**Describe the bug**

If the Markdown document contains:

```md
If Alice gives Bob $10, and Bob gives Eve $10, how much money does Bob have?
```

Then, unless `myst_enable_extensions = []` is set in `conf.py`, the HTML output is:

```html
<p>If Alice gives Bob <span class="math notranslate nohighlight">\(10, and Bob gives Eve \)</span>10, how much money does Bob have?</p>
```

**To Reproduce**

1. Clone https://github.com/jpmckinney/myst-parser-tests
2. Run instructions at https://github.com/jpmckinney/myst-parser-tests/blob/master/index.md

**Expected behavior**

A less surprising result would be:

```html
<p>If Alice gives Bob $10, and Bob gives Eve $10, how much money does Bob have?</p>
```

I don't know MathJax as I've never used it. Maybe it's impossible to determine that `$10, and Bob gives Eve $` is not a math string. If that's the case, it might be a good idea to not include `dollarmath` by default.

**Comment by choldgraf** [comment]
I think this is gonna be a tricky one depending on the use-case. In scientific fields it is super common to use dollar math for equations, which is why this is turned on by default (since MyST grew out of Jupyter Book, and that is a data science-specific use-case).

To get around this, you can just "escape" the dollar signs. Can you try the following:

```
If Alice gives Bob \$10, and Bob gives Eve \$10, how much money does Bob have?
```

?

**Comment by chrisjsewell** [comment]
I suggest you re-read https://myst-parser.readthedocs.io/en/latest/using/syntax.html#dollar-delimited-math 😉: This is exactly what `myst_dmath_allow_space` and `myst_dmath_allow_digits` are for

I'm open to discussion, as to whether or not these should be `False` by default

**Comment by jpmckinney** [comment]
For my use case, I'm fine with just setting `myst_enable_extensions = []`, to guarantee that dollarmath is never run. I don't know how changing the defaults of those two other variables will affect users who do want dollarmath to run sometimes, so I'll leave it to you to consider appropriate defaults.

**Comment by chrisjsewell** [comment]
> to guarantee that dollarmath is never run

yeh no worries, it obviously just a bit problematic to remove dollarmath completely as a default now, which might break a lot of peoples docs

**Comment by choldgraf** [comment]
Hmmm, should we close this one? I don't think it's actionable since we're not going to change the default. Do we need to document something more clearly?

**Comment by jpmckinney** [comment]
All good for me - just wanted to check the defaults :)

**Comment by jpmckinney** [comment]
🙏 Fixed by #505
